HPCC-15456 Add units to file sizes

kunalaswani opened the pull request:
Added conversion of units to bytes.

Signed-off-by: Aswani, Kunal <kunal.aswani@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browsers
  - [x] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

kunalaswani committed:
HPCC-15456 Add units to file sizes

Added conversion of units to bytes.

Signed-off-by: Aswani, Kunal <kunal.aswani@lexisnexis.com>

kunalaswani commented:
@GordonSmith please review

miguelvazq commented on esp/src/eclwatch/DFUQueryWidget.js:
```diff
@@ -632,12 +632,14 @@ define([
-                                if (intsize === null) {
-                                    return 0
+                                if(intsize === null ) {
+                                    return '0 Bytes';
```
This needs translating?

GordonSmith replied:
Should be fine.

miguelvazq commented on esp/src/eclwatch/LZBrowseWidget.js:
```diff
@@ -676,7 +676,9 @@ define([
                                     return ""
                                 }
-                                return fsize.toLocaleString();
+                                var unitConversion = ["Bytes","KB","MB","GB","TB","PB","EB","ZB","YB"],
```
No 'Zero bytes' here?

kunalaswani replied:
This is in the Landing zone, prior to conversion of unit size, we did not put a size measurement in the dropzone / node ip/ directory level, those return blank as prior. 

Not sure if we want that to change, or what should be written there.

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 9cf80ab898a56f24322813936c2f5478ae9d3ae0
Build: success
ECL Watch:  Rebuilding Site
Rebuild: success
Install hpccsystems-platform-community_7.0.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |787 |787 |0 |
|test (thor) |718 |718 |0 |
|test (roxie) |831 |829 |2 |


[Errors:](https://gist.github.com/50753f33f028be63565b3b9d3658130b)
- test
  - roxie
    - 559\. [roxiepipe](https://gist.github.com/HPCCSmoketest/50753f33f028be63565b3b9d3658130b#file-roxiepipe-w20180614-191543-2-log), 604\. [soapcall (targetIP=127001, goodPort=9876, blacListedPort=9875)](https://gist.github.com/HPCCSmoketest/50753f33f028be63565b3b9d3658130b#file-soapcall-targetip-127001,-goodport-9876,-blaclistedport-9875-w20180614-191600-2-log), 

HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   41 sec (00:00:41) |  173 sec (00:02:53) |  128 sec (00:02:08) |   10 sec (00:00:10) |   30 sec (00:00:30) | 1383 sec (00:23:03) |   17 sec (00:00:17) | 1782 sec (00:29:42) |

AttilaVamos commented:
The Roxie errors are not related to this PR.

GordonSmith commented on esp/src/eclwatch/DFUQueryWidget.js:
```diff
@@ -632,12 +632,14 @@ define([
-                                    return intsize.toLocaleString();
+                                var unitConversion = ["Bytes","KB","MB","GB","TB","PB","EB","ZB","YB"],
+                                x = Math.floor(Math.log(intsize)/ Math.log(1024));
+                                return (intsize/ Math.pow(1024,x)).toFixed(2) + ' ' + unitConversion[x];
```
This code should be moved into Utility as a new function.

GordonSmith reviewed (changes requested):
@AttilaVamos I was expecting the linter to complain about the use of single quotes (instead of double).

kunalaswani commented:
@GordonSmith Function moved to utility file, and updated. please review.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 08a6f84f3c7b49118dc77ece39cfa3bc724d4516
Build: success
ECL Watch:  Rebuilding Site
Rebuild: success
Install hpccsystems-platform-community_7.0.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |787 |787 |0 |
|test (thor) |718 |718 |0 |
|test (roxie) |831 |831 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   38 sec (00:00:38) |  171 sec (00:02:51) |  132 sec (00:02:12) |    9 sec (00:00:09) |   30 sec (00:00:30) | 1354 sec (00:22:34) |   18 sec (00:00:18) | 1752 sec (00:29:12) |

GordonSmith commented on esp/src/src/Utility.ts:
```diff
@@ -91,6 +91,11 @@ export function espTime2SecondsTests() {
 }
 
+export function convertedSize (intsize){
+    var unitConversion = ["Bytes","KB","MB","GB","TB","PB","EB","ZB","YB"], x = Math.floor(Math.log(intsize)/ Math.log(1024));
```
Minor: Keep one var declaration per line for consistency:
```
var unitConversion = ["Bytes","KB","MB","GB","TB","PB","EB","ZB","YB"];
var x = Math.floor(Math.log(intsize)/ Math.log(1024));

```

GordonSmith commented on esp/src/src/Utility.ts:
```diff
@@ -91,6 +91,11 @@ export function espTime2SecondsTests() {
+    var unitConversion = ["Bytes","KB","MB","GB","TB","PB","EB","ZB","YB"], x = Math.floor(Math.log(intsize)/ Math.log(1024));
+    return (intsize/ Math.pow(1024,x)).toFixed(2) + " " + unitConversion[x];
+}
+
```
FYI:  Since this is TypeScript we can improve the overall declaration by adding some types:
```typescript
export function convertedSize(intsize: number): string {
    const unitConversion = ["Bytes", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB"];
    const x = Math.floor(Math.log(intsize) / Math.log(1024));
    return (intsize / Math.pow(1024, x)).toFixed(2) + " " + unitConversion[x];
}
```
Type of thing.

kunalaswani commented:
@GordonSmith I changed them from var to const, I had tested with const earlier and wasn't sure which was the preferred method, and I have moved the other const to a new line. I have retested and the numbers are converting correctly.

GordonSmith reviewed (approved):
Looks good.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: b5a874bc9fb570c70bd89995b19c6ecdfdfd98e8
Build: success
ECL Watch:  Rebuilding Site
Rebuild: success
Install hpccsystems-platform-community_7.0.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |787 |787 |0 |
|test (thor) |718 |718 |0 |
|test (roxie) |831 |831 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   22 sec (00:00:22) |  171 sec (00:02:51) |  125 sec (00:02:05) |    8 sec (00:00:08) |   29 sec (00:00:29) | 1228 sec (00:20:28) |   17 sec (00:00:17) | 1600 sec (00:26:40) |

kunalaswani commented:
@richardkchapman please merge
